### DISCUSSION TITLE
how use the result of DCT scan in csv format to make tesr file or tess file in Neper?

### DISCUSSION THREAD

**Discussion by Mahdieh-karami** [neperfepx/neper#906] [Q&A] [answered]
Hi, I have a csv file which include grain ID , equivalent diameter of grains , centroids and orientation . Does any bode know how this data can be used to make a polycrystalline model for CPFE? I know I can make tesr file in Neper according https://neper.info/tutorials/morpho_tesr_mesh.html#generating-a-clean-raster-tessellation-tesr  but  I have problem the Neper can not reed my file . I will be thankful if anybody can help me .

**Comment by rquey** [comment, marked as answer]
This can be done by writing a file for the grain centroids and equivalent diameters, and a separate file for the orientations. You also need to define the "domain". See [-domain](https://neper.info/doc/neper_t.html#cmdoption-domain), [-morpho](https://neper.info/doc/neper_t.html#cmdoption-morpho), [-ori](https://neper.info/doc/neper_t.html#cmdoption-ori), and also [Data File](https://neper.info/doc/fileformat.html#data-file).

The syntax is:
```
neper -T -n <grain_number> -domain "<domain>" -morpho "centroiddiameq:file(<centroid_diameq_file>)" -ori file (<orientation_file>)
```

**Reply by Mahdieh-karami** [reply]
Thank you so much for the guidance, it fixed my issue!
